ChangeLog.d/fix-printf-specifiers.txt

```diff
@@ -0,0 +1,10 @@
+Bugfix
+   * Add printf function attributes to mbedtls_debug_print_msg because we were
+     not actually getting any printf format specifier warnings prior to this.
+     Also add extra printf compiler warning flags to builds.
+Requirement changes
+   * Use %zu for print of type size_t in new work, however please do not use
```

Comment:
The target reader of ChangeLog entries is normally a basic user. I think this entry should be rewritten to start with that perspective before shifting to the perspective of a developer or contributor. Maybe something like "The library now uses the %zu format specifier with the printf() family of functions, so requires a toolchain that supports it. This change does not affect the maintained LTS branches, so when contributing changes please [...]"